### ISSUE TITLE
Historical feature retrieval: how to join two tables with `.id` field?

### ISSUE THREAD

**Issue by Felix-neko** [feast-dev/feast#2212]
Hi folks!
I want to join and extract data from two tables both of them has an id  column as a primary key
(e.g. `buyer.id` and `house.id`).

How can I tell Feast that I want to get historical features with
`[{"house:id": 1, "buyer:id": 2, "event_timestamp": now()}]` ?

**Comment by tsotnet** [comment]
I'm not sure this is possible on the `get_historical_features` level. However, in most cases you can rename columns on the data source level (e.g. `BigQuerySource` and `RedshiftSource` both support `query` parameter in which you can write a SELECT SQL query where you can rename columns). So for example, in house table you can select id column as house_id and similarly in the buyer table select id column as buyer_id. As far as Feast is concerned, it'll only know about house_id and buyer_id, and not id, so you'll be able to query for `[{"house_id": 1, "buyer_id": 2, "event_timestamp": now()}]`

**Comment by adchia** [comment]
Yeah in the get_historical_features logic for BigQuery/Redshift, from scanning the code it looks like we expect entity_df to have the join keys, so if you don't have unique join keys, you'd have to do something like what @tsotnet said
